### PR TITLE
refactor(credit note): improve modal UI

### LINKED ISSUE #767: (proposal) Replace CREDIT_NOTE transaction type with generic REVERSAL
I've recently completed some work on credit notes (see #764).  We currently use a `CREDIT_NOTE` voucher type to signal that an invoice is reversed.  If we were to expand this process to Cash Payments and vouchers, we would need to have `CREDIT_NOTE` and `VOUCHER_NOTE` or something like that.

I propose that we just make a generic `REVERSAL` flag.  This would allow us to find records that have been reversed very easily in the posting journal alone, and allow use to filter those records out of the `voucher` registry easily.  It also reduces the need for logic such as `if (type === CREDIT_NOTE || type === VOUCHER_NOTE || ...)` for reporting/filtering.

### PR DESCRIPTION
**Pull request by jniles**
This commit improves the credit note modal user interface.  It gives more context about the bill and removes the HTML tables for layout.  It also attempts to improve the dialogs' text to aid users in making the right decision.

Finally, the "REVERSAL" text has been changed to "CORRECTION" since that is both French and English and is easier to refer to.

Closes #767.

**Screenshots**
creditnoteimproveddialog
_Fig 1: New English Dialog Box_

newcreditnotemodalinterfacenormal
_Fig 2: New Credit Note Modal Interface_

### CODE REVIEW

File: client/src/i18n/fr/form.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
             "CONFIRM_ACTION": "Confirmez vous cette action ?",
             "CONFIRMATION_NEEDED": "Confirmation Requise",
             "CONFIRM_ACTIVATION": "Voulez vous vraiment activer les droits d'accès de cet utilisateur",
-            "CONFIRM_CREDIT_NOTE": "La facture dont vous voulez annuler a déjà été payée, voulez-vous vraiment annuler cette facture ",
+            "CONFIRM_CREDIT_NOTE": "Attention! La facture dont vous voulez annuler a déjà été payée! Voulez-vous vraiment annuler cette facture?",
```
Comment (mbayopanda) [inline client/src/i18n/fr/form.json:82]: The correct corresponding translation in French is : 
`Attention! La facture que vous avez sélectionnée a déjà été payée! L'inverser peut entraîner des déséquilibres dans votre comptabilité. Êtes-vous sûr de vouloir continuer à annuler cette facture?`

File: client/src/i18n/fr/patient_invoice.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
         "STAT_PAID": "Payés",
         "STAT_NOT_PAID": "En attente de paiement",
         "SUBSIDY_BREAKDOWN": "Subvention disfonctionnelle",
-        "SUCCESS": "Facture des patients enregistrés avec succès."
+        "SUCCESS": "Facture des patients enregistrés avec succès.",
+        "ALERT_CREDIT_NOTE" : "Vous annulez la facture <strong>{{ invoiceReference }}</strong> de <u><span class='text-capitalize'>{{ patientName }}</span> ({{ patientReference }})</u> avec un valeur total de {{ cost }}. Veuillez vous assurer que les informations ci-dessous sont correctes. Cette action n'est pas réversible."
```
Comment (mbayopanda) [inline client/src/i18n/fr/patient_invoice.json:25]: The correct phrase is : 
`Vous annulez la facture <strong>{{ invoiceReference }}</strong> de <u><span class='text-capitalize'>{{ patientName }}</span> ({{ patientReference }})</u> avec une valeur totale de {{ cost }}. Veuillez vous assurer que les informations ci-dessous sont correctes. Cette action n'est pas réversible.
`

File: client/src/modules/invoices/registry/modalCreditNote.js [hunk 3]
```diff
@@ -2,58 +2,80 @@ angular.module('bhima.controllers')
...
 
-      vm.billingAmount = sum(response.billing, 'value');
-      vm.subsidyAmount = sum(response.subsidy, 'value');
-   
+      vm.billingAmount = response.billing.reduce((a, b) => a + b.value, 0);
+      vm.subsidyAmount = response.subsidy.reduce((a, b) => a + b.value, 0);
+
+      vm.hasBillingServices = vm.billingAmount > 0;
+      vm.hasSubsidies = vm.subsidyAmount > 0;
+      vm.hasExpandedFooter = vm.hasBillingServices || vm.hasSubsidies;
```
Comment (mbayopanda) [inline client/src/modules/invoices/registry/modalCreditNote.js:40]: 👍 this section is very informative

File: client/src/modules/invoices/registry/registry.js [hunk 4]
```diff
@@ -117,38 +114,35 @@ function InvoiceRegistryController(
...
           invoice._backgroundColor = invoice.reversed ? reversedBackgroundColor : regularBackgroundColor;
-          invoice._is_cancelled = invoice.reversed;
         });
 
         // put data in the grid
         vm.uiGridOptions.data = invoices;
       })
       .catch(handler)
-      .finally(function () {
+      .finally(() => {
```
Comment (mbayopanda) [inline client/src/modules/invoices/registry/registry.js:135]: is the inline syntax doesn't work ? 
`.finally(toggleLoadingIndicator);`

It is more simple and nice to see.

File: server/controllers/finance/patientInvoice.js [hunk 5]
```diff
@@ -274,10 +274,8 @@ function lookupInvoiceCreditNote(invoiceUuid) {
     JOIN user u ON u.id = v.user_id
     JOIN invoice i ON i.uuid = v.reference_uuid
     WHERE v.type_id = ${CREDIT_NOTE_ID} AND v.reference_uuid = ?`;
+
   return db.one(sql, [buid])
-    .then(creditNote => {
```
Comment (mbayopanda) [inline server/controllers/finance/patientInvoice.js:278]: Good improvement.

File: server/models/procedures/voucher.sql [hunk 6]
```diff
@@ -125,7 +125,7 @@ BEGIN
   -- @todo - make only one type of reversal (not cash, credit, or voucher)
 
   INSERT INTO voucher (uuid, date, project_id, currency_id, amount, description, user_id, type_id, reference_uuid)
-    SELECT voucher_uuid, NOW(), zz.project_id, enterprise.currency_id, 0, CONCAT_WS(' ', '(REVERSAL)', description), user_id, reversalType, uuid
+    SELECT voucher_uuid, NOW(), zz.project_id, enterprise.currency_id, 0, CONCAT_WS(' ', '(CORRECTION)', description), user_id, reversalType, uuid
```
Comment (mbayopanda) [inline server/models/procedures/voucher.sql:128]: Good name, it works for both English and French, and it is more meaningful.

**Review by mbayopanda — APPROVED** [review]
I made some non-blocking comments about this PR.

The code is excellent as always in your habits.

LGTM 👍

File: client/src/modules/invoices/registry/registry.js (hunk 4 above)
Comment (jniles) [reply client/src/modules/invoices/registry/registry.js:135]: Your suggestion will work fine here.  We can start doing it 👍

File: server/controllers/finance/patientInvoice.js (hunk 5 above)
Comment (jniles) [reply server/controllers/finance/patientInvoice.js:278]: 💃

**Pull request merged**